src/main/java/org/graylog/collector/inputs/InputConfiguration.java

```diff
@@ -40,12 +46,39 @@
     @NotNull
     private Set<String> outputs = Sets.newHashSet();
 
+    private final MessageFields messageFields = new MessageFields();
+
     public InputConfiguration(String id, Config config) {
         this.id = id;
 
         if (config.hasPath("outputs")) {
             this.outputs = Sets.newHashSet(Splitter.on(",").omitEmptyStrings().trimResults().split(config.getString("outputs")));
         }
+
+        if (config.hasPath("message-fields")) {
+            final Config messageFieldsConfig = config.getConfig("message-fields");
+
+            for (Map.Entry<String, ConfigValue> entry : messageFieldsConfig.entrySet()) {
+                final String key = entry.getKey();
+                final ConfigValue value = entry.getValue();
+
+                switch (value.valueType()) {
+                    case NUMBER:
+                        this.messageFields.put(key, messageFieldsConfig.getLong(key));
+                        break;
+                    case BOOLEAN:
+                        this.messageFields.put(key, messageFieldsConfig.getBoolean(key));
```

Comment:
I know, but once this is fixed on the server, it will just work. :wink: 

Same here btw: https://github.com/Graylog2/gelfclient/blob/master/src/main/java/org/graylog2/gelfclient/encoder/GelfMessageJsonEncoder.java#L97-L104